### ISSUE TITLE
Remote Docker Credential Helper Fails

### ISSUE THREAD

**Issue by mhosmar-cpr** [microsoft/vscode-remote-release#7871]
- VSCode Version: 1.74.3
- Local OS Version: Win 10
- Remote OS Version: Ubuntu Focal
- Remote Extension/Connection Type: Devcontainer WSL2 Docker Desktop

Getting a failure when trying to pull a docker image from a private registry from within a dev container using docker-from-docker. Relates to #7184.

Output when trying to add credentials with docker login:
`Error saving credentials: error storing credentials - err: exit status 255, out: ``
`

Steps to Reproduce:

1. log into private registry on host machine (using pwsh)
3. Use scripts found here to enable docker-from-docker [docker-from-docker](https://github.com/microsoft/vscode-dev-containers/tree/main/containers/docker-from-docker)
4. Use the ubuntu:focal image as the base image for dev container
5. Reopen in container
6. Attempt to pull from private registry from within container. -- "Not authorized"
7. Attempt to login to private registry from within container. -- Above response

**Comment by chrmarti** [comment]
Which registry are you using? Could you append the log from when you try this? (`F1` > `Dev Containers: Show Container Log`)

**Comment by mhosmar-cpr** [comment]
It is a private github container registry. Do you just need the log after running a pull? I can include the whole creation as well if needed.

This is after using docker compose pull.

```
[3791105 ms] Start: Run in container: cat /proc/12796/environ
[3799536 ms] Start: Run in container: cat /proc/12915/environ
[3799782 ms] Container server: {
  message: 'write EPIPE',
  name: 'Error',
  stack: 'Error: write EPIPE\n' +
    '\tat afterWriteDispatched (node:internal/stream_base_commons:160:15)\n' +
    '\tat writeGeneric (node:internal/stream_base_commons:151:3)\n' +
    '\tat Socket._writeGeneric (node:net:795:11)\n' +
    '\tat Socket._write (node:net:807:8)\n' +
    '\tat writeOrBuffer (node:internal/streams/writable:389:12)\n' +
    '\tat _write (node:internal/streams/writable:330:10)\n' +
    '\tat Writable.write (node:internal/streams/writable:334:10)\n' +
    '\tat c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:87:219\n' +
    '\tat new Promise (<anonymous>)\n' +
    '\tat Fe (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:86:1504)\n' +
    '\tat async vk (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:67:709)\n' +
    '\tat async yk (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:67:79)\n' +
    '\tat async Object.rpc (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:83:901)'
}
[3799783 ms] Container server: {
  message: 'write EPIPE',
  name: 'Error',
  stack: 'Error: write EPIPE\n' +
    '\tat afterWriteDispatched (node:internal/stream_base_commons:160:15)\n' +
    '\tat writeGeneric (node:internal/stream_base_commons:151:3)\n' +
    '\tat Socket._writeGeneric (node:net:795:11)\n' +
    '\tat Socket._write (node:net:807:8)\n' +
    '\tat writeOrBuffer (node:internal/streams/writable:389:12)\n' +
    '\tat _write (node:internal/streams/writable:330:10)\n' +
    '\tat Writable.write (node:internal/streams/writable:334:10)\n' +
    '\tat c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:87:219\n' +
    '\tat new Promise (<anonymous>)\n' +
    '\tat Fe (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:86:1504)\n' +
    '\tat async vk (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:67:709)\n' +
    '\tat async yk (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:67:79)\n' +
    '\tat async Object.rpc (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:83:901)'
}
{
  message: 'write EPIPE',
  name: 'Error',
  stack: 'Error: write EPIPE\n' +
    '\tat afterWriteDispatched (node:internal/stream_base_commons:160:15)\n' +
    '\tat writeGeneric (node:internal/stream_base_commons:151:3)\n' +
    '\tat Socket._writeGeneric (node:net:795:11)\n' +
    '\tat Socket._write (node:net:807:8)\n' +
    '\tat writeOrBuffer (node:internal/streams/writable:389:12)\n' +
    '\tat _write (node:internal/streams/writable:330:10)\n' +
    '\tat Writable.write (node:internal/streams/writable:334:10)\n' +
    '\tat c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:87:219\n' +
    '\tat new Promise (<anonymous>)\n' +
    '\tat Fe (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:86:1504)\n' +
    '\tat async vk (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:67:709)\n' +
    '\tat async yk (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:67:79)\n' +
    '\tat async Object.rpc (c:\\Users\\mikeh\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.269.0\\dist\\extension\\extension.js:83:901)'
}
```

**Comment by chrmarti** [comment]
Fixed in https://github.com/microsoft/vscode-remote-release/issues/7854 and available as Dev Containers extension 0.273.0-pre-release.
